### PR TITLE
cython 3.0.6

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-3683](https://anaconda.atlassian.net/browse/PKG-3683) 
- [Upstream repository](https://github.com/cython/cython/tree/3.0.6)
- Upstream requirements: 
  - https://github.com/cython/cython/blob/3.0.6/setup.py

### Explanation of changes:

- Add `setuptools  # [py>=312]` as a test dependency for `distutils`

### Notes:

- gevent 23.9.1 requires cython >=3.0.2 https://github.com/AnacondaRecipes/gevent-feedstock/pull/5

[PKG-3683]: https://anaconda.atlassian.net/browse/PKG-3683?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,16 +29,18 @@ requirements:
     - python
 
 test:
+  files:
+    - fib.pyx
```
Comment (danpetry) [inline recipe/meta.yaml:33]: fib.pyx was used for run_test.py, which isn't being used anymore. could it be reinstated or (better) upstream tests run?

**Review by danpetry — APPROVED** [review]
test improvement would be good, but otherwise fine

File: recipe/meta.yaml [hunk 2]
```diff
@@ -29,16 +29,18 @@ requirements:
...
   requires:
     - pip
+    # for distutils
+    - setuptools  # [py>=312]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: Why is this needed here but not in the runtime section if the tests aren't being run?

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:33]: I've rewritten run_test.py to use fib.pyx. The upstream tests are too slow.

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Yeah, it was needed for running tests and run_test.py. I'll add tests but they are quite slow

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Tests are too (_extremely_) slow. I added a comment about them. But `setuptools` is still needed for `run_test.py` in the try/except block

File: recipe/run_test.py [hunk 3]
```diff
@@ -34,24 +33,26 @@
...
+pprint('PATH: %r' % os.environ['PATH'])
+pprint('CWD: %r' % os.getcwd())
 
-from distutils.spawn import find_executable
-from distutils.core import setup
-from distutils.extension import Extension
+try:
+    from setuptools import setup, Extension
+except ImportError:
+    from distutils.core import setup, Extension
```
Comment (JeanChristopheMorinPerso) [inline recipe/run_test.py]: I would just use setuptools. There is no reason to rely directly on distutils.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: recipe/run_test.py (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/run_test.py]: Done
